packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -248,89 +284,122 @@ export class HostedPluginSupport {
-            this.initPlugins(manager, {
-                ...initData,
-                plugins: hostContributions.map(contributions => contributions.plugin)
-            }).then(() => {
-                if (toDisconnect.disposed) {
-                    return;
-                }
-                for (const contributions of hostContributions) {
-                    const plugin = contributions.plugin;
-                    const id = plugin.model.id;
-                    contributions.state = PluginContributions.State.STARTED;
-                    console.log(`[${id}]: Started plugin.`);
-                    toDisconnect.push(contributions.push(Disposable.create(() => {
-                        console.log(`[${id}]: Stopped plugin.`);
-                        manager.$stop(id);
-                    })));
-
-                    this.activateByWorkspaceContains(manager, plugin);
-                }
+            const manager = await this.obtainManager(host, hostContributions, toDisconnect);
+            if (!manager) {
+                return;
+            }
+            const plugins = hostContributions.map(contributions => {
+                const metadata = {
+                    ...contributions.plugin,
+                    model: {
+                        ...contributions.plugin.model
+                    }
+                };
+                // remove contributions info in order to reduce size of JSON-RPC messages
+                delete metadata.model.contributes;
+                delete metadata.model.activationEvents;
+                return metadata;
             });
+            thenable.push((async () => {
+                try {
+                    const activationEvents = [...this.activationEvents];
+                    await manager.$start({ plugins, configStorage, activationEvents });
+                    if (toDisconnect.disposed) {
+                        return;
+                    }
+                    for (const contributions of hostContributions) {
+                        started++;
+                        const plugin = contributions.plugin;
+                        const id = plugin.model.id;
+                        contributions.state = PluginContributions.State.STARTED;
+                        console.log(`[${this.clientId}][${id}]: Started plugin.`);
+                        toDisconnect.push(contributions.push(Disposable.create(() => {
+                            console.log(`[${this.clientId}][${id}]: Stopped plugin.`);
+                            manager.$stop(id);
+                        })));
+
+                        this.activateByWorkspaceContains(manager, plugin);
+                    }
+                } catch (e) {
+                    console.error(`Failed to start plugins for '${host}' host`, e);
+                }
+            })());
         }
-        return toDisconnect;
+        await Promise.all(thenable);
+
+        const pluginCount = `${started} plugin${started === 1 ? '' : 's'}`;
```

Comment:
FYI there is https://www.npmjs.com/package/pluralize

Reply:
Do we need 500 lines of an external package to replace one liner?

Reply:
it's just that I see it 3 times only in that file

Reply:
will remove duplication

Reply:
extracted `logMeasurement` method